### PR TITLE
Feat: implement createGroup api client

### PR DESCRIPTION
**Pull request by kanta1207**
## Overview

- Related issue : #181
- Implement `createGroup` api client, which works in client component
- Invoke `createGroup` method in `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx` with pure html `form` tag for now.
## Changes

- Add `createGroup` method to `src/lib/api/group/client/groupApiClient.client.ts`
- Add html `<form>` tag to `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx`
- Invoke `createGroup` in `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx`

## What I didn't in this PR
**I will tackle on Implementation of `CreateGroupForm` component in next scope, using common `Form` component and react-hook-form.** 
In this PR, I only focused on the API call.

## Review points

- Is API client `createGroup` method properly implemented with its test

## Assignee Checklist:

- [x] The base branch is correct (See: [Types of Branches](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] Code is well-commented and easy to understand
- [ ] UI changes accurately reflect the provided design

### CODE REVIEW

**Comment by kanta1207** [comment]
Importing `createGroupAction' into client component causing a build error.

**Review by kotaaaa — APPROVED** [review]
I thought this PR's scope is a little bit small, but it's good to go. Thanks!

File: src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx [hunk 1]
```diff
@@ -8,34 +8,45 @@ import {
...
+import { createGroup } from '@/lib/api/group/client';
 
-import { FC } from 'react';
+import { FormEvent, useState } from 'react';
 
-interface ICreateGroupDrawerContentProps {
-  handleCreateGroup: () => void;
-}
+export const CreateGroupDrawerContent = () => {
+  const [groupName, setGroupName] = useState('');
```
Comment (uskayyyyy) [inline src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx:16]: Could you add `<IGroup['name']>`, please?
Currently, the type is inferred as a simple `string`, not referring to the type of group name.

File: src/lib/api/group/client/groupApiClient.client.ts [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
-//  */
-// export interface ICreateGroupApiResponse {
-//   groupId: string;
-// }
+/**
+ * Interface representing the API response for the method to create a group.
+ * @property {string} groupId - The unique identifier of the group.
+ */
+export interface ICreateGroupApiResponse {
+  groupId: string;
```
Comment (uskayyyyy) [inline src/lib/api/group/client/groupApiClient.client.ts:19]: `IGroup['id']`

File: src/lib/api/group/client/groupApiClient.client.ts [hunk 3]
```diff
@@ -1,39 +1,38 @@
...
-//  */
-// export interface ICreateGroupParams {
-//   groupName: string;
-// }
+/**
+ * Interface representing the Params for the method to create a group.
+ * @property {string} groupName - The name of the group.
+ */
+export interface ICreateGroupParams {
+  groupName: string;
```
Comment (uskayyyyy) [inline src/lib/api/group/client/groupApiClient.client.ts:11]: `IGroup['name']`

**Review by uskayyyyy — CHANGES_REQUESTED** [review]
@kanta1207 
Thank you for the implementation! Please kindly check the comments I left.

**Comment by kanta1207** [comment]
> @kanta1207 Thank you for the implementation! Please kindly check the comments I left.

Thank you for your review! I addressed the points you brought up. Please take a relook at them 🙏

**Review by uskayyyyy — APPROVED** [review]
@kanta1207 
Thank you for updating! LGTM :)

**Comment by kanta1207** [comment]
> I thought this PR's scope is a little bit small, but it's good to go. Thanks!

@kotaaaa Yeah indeed maybe I tried to make it too small 😅 
I'll just replace `form` with `Form` component quickly in next PR

**Pull request merged**